java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -86,141 +65,44 @@ public VisitorsBridge(JavaFileScanner visitor, List<File> projectClasspath) {
-      }
-    }
+  public TestJavaFileScannerContext lastCreatedTestContext() {
+    return testContext;
   }
 
-  public VisitorContext getContext() {
-    return context;
-  }
+  public static class TestJavaFileScannerContext extends DefaultJavaFileScannerContext {
 
-  public void setContext(VisitorContext context) {
-    this.context = context;
-  }
+    private final Set<AnalyzerMessage> issues = new HashSet<>();
 
-  @VisibleForTesting
-  public static class DefaultJavaFileScannerContext implements JavaFileScannerContext {
-    private final CompilationUnitTree tree;
-    @VisibleForTesting
-    public final SourceFile sourceFile;
-    private final SemanticModel semanticModel;
-    private final SonarComponents sonarComponents;
-    private final ComplexityVisitor complexityVisitor;
-    private final File file;
-
-    public DefaultJavaFileScannerContext(
+    public TestJavaFileScannerContext(
       CompilationUnitTree tree, SourceFile sourceFile, File file, SemanticModel semanticModel, boolean analyseAccessors, @Nullable SonarComponents sonarComponents) {
-      this.tree = tree;
-      this.sourceFile = sourceFile;
-      this.file = file;
-      this.semanticModel = semanticModel;
-      this.sonarComponents = sonarComponents;
-      this.complexityVisitor = new ComplexityVisitor(analyseAccessors);
-    }
-
-    @Override
-    public CompilationUnitTree getTree() {
-      return tree;
+      super(tree, sourceFile, file, semanticModel, analyseAccessors, sonarComponents);
     }
 
-    @Override
-    public void addIssue(Tree tree, JavaCheck javaCheck, String message) {
-      addIssue(((JavaTree) tree).getLine(), javaCheck, message, null);
-    }
-
-    @Override
-    public void addIssue(Tree tree, JavaCheck check, String message, @Nullable Double cost) {
-      addIssue(((JavaTree) tree).getLine(), check, message, cost);
+    public Set<AnalyzerMessage> getIssues() {
+      return issues;
     }
 
     @Override
-    public void addIssueOnFile(JavaCheck javaCheck, String message) {
-      addIssue(-1, javaCheck, message);
-    }
-
-    @Override
-    public void addIssue(int line, JavaCheck javaCheck, String message) {
-      addIssue(line, javaCheck, message, null);
+    public void addIssue(int line, JavaCheck javaCheck, String message, @Nullable Double cost) {
+      issues.add(new AnalyzerMessage(javaCheck, getFile(), line, message, cost != null ? cost.intValue() : 0));
+      addIssueForCheckMessageVerifier(line, javaCheck, message, cost);
     }
 
-    @Override
-    public void addIssue(int line, JavaCheck javaCheck, String message, @Nullable Double cost) {
-      Preconditions.checkNotNull(javaCheck);
-      Preconditions.checkNotNull(message);
+    /**
+     * @deprecated should be removed once CheckMessageVerifier support is dropped
+     */
+    @Deprecated
+    private void addIssueForCheckMessageVerifier(int line, JavaCheck javaCheck, String message, @Nullable Double cost) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)